build-tools/scripts/jenkinsfile

```diff
@@ -0,0 +1,484 @@
+// This file is based on the Jenkins declarative pipeline syntax (as opposed to the scripted pipeline syntax)
+// https://jenkins.io/doc/book/pipeline/syntax/
+// https://jenkins.io/blog/2017/02/03/declarative-pipeline-ga/
+
+def botLabel = 'xamarin-android-master-pipeline-eval'
+
+def XADir = "xamarin-android"
+
+def publishBuildFilePaths = null
+def publishTestFilePaths = null
+
+def MSBUILD_AUTOPROVISION_ARGS="/p:AutoProvision=True /p:AutoProvisionUsesSudo=True /p:IgnoreMaxMonoVersion=False"
+
+def buildStatusProcessed = false
+def buildStatusUploaded = false
+
+def isPR = false                // Default to CI
+
+def processBuildStatus() {
+    def status = 0
+
+    try {
+        echo "processing build status"
+        sh "make package-build-status CONFIGURATION=${env.BuildFlavor}"
+    } catch (error) {
+        echo "ERROR : NON-FATAL : processBuildStatus: Unexpected error: ${error}"
+        status = 1
+    }
+
+    return status
+}
+
+def publishPackages(filePaths) {
+    def status = 0
+    try {
+        echo "Publishing to Azure storage account: https://ms.portal.azure.com/#@microsoft.onmicrosoft.com/resource/subscriptions/7b4817ae-218f-464a-bab1-a9df2d99e1e5/resourceGroups/Group/providers/Microsoft.ClassicStorage/storageAccounts/xamjenkinsartifact/overview"
+        step([$class: 'WAStoragePublisher', allowAnonymousAccess: true, cleanUpContainer: false, cntPubAccess: true, containerName: "${env.ContainerName}", doNotFailIfArchivingReturnsNothing: false,
```

Comment:
Also in the interest of "consistency", is there an `azureUpload()` method we should be using, a'la https://github.com/mono/mono/commit/b538187 ?

(Also, how's ^^ work?  I don't see a definition for `azureUpload()` in that file or that repo.)

Reply:
Yes you should be using `azureUpload`. It's provided by the Azure Storage Jenkins plugin.

Reply:
I spoke with @akoeplinger. I'm going to update to using this function provided by the Azure Blob Jenkins plugin. I've put the change in my test branch and am testing it now.